src/transformers/trainer.py

```diff
@@ -233,6 +243,117 @@ def __init__(
         if self.args.fp16 and _use_native_amp:
             self.scaler = torch.cuda.amp.GradScaler()
 
+    @classmethod
+    def from_nlp_dataset(
+        cls,
+        model: PreTrainedModel,
+        args: TrainingArguments,
+        dataset_dict: "nlp.dataset_dict.DatasetDict" = None,
+        train_dataset: "nlp.dataset_dict.Dataset" = None,
+        eval_dataset: "nlp.dataset_dict.Dataset" = None,
+        data_collator: Optional[DataCollator] = None,
+        metrics: Optional[Union["nlp.Metric", List["nlp.Metric"]]] = None,
+        final_activation: Optional[Union[str, FinalActivation, Callable]] = None,
```

Comment:
I think I would call this `metrics_preprocessing` and accept either a callable or an activation string/enum